Source/DataSources/KmlDataSource.js

```diff
@@ -1688,11 +2213,14 @@ define([
      * @param {Object} [options] An object with the following properties:
      * @param {DefaultProxy} [options.proxy] A proxy to be used for loading external data.
      * @param {String} [options.sourceUri] Overrides the url to use for resolving relative links and other KML network features.
+     * @param {Camera} [options.camera] The camera that is used for viewRefreshModes and sending camera properties to network links.
+     * @param {Canvas} [options.canvas] The canvas that is used for sending viewer properties to network links.
```

Comment:
I'm really torn on camera and canvas being optional.  The idealist in me says make them optional, but I think we have to require them for practical reasons.  We can detect and spit out a deprecation message for a few releases if they are missing and then start throwing as part of 1.21.

My concern is that most defaults would end up shipping apps that can't load NetworkLinks and find out about them too late.  In the past (Insight3D), we make certain parts of KML optional (like MSL), and this happened all of the time.  A user would build and ship an app testing with basic KML files.  Their users then load a KML file that requires the optional parameters and the KML doesn't load correctly.

Also, since they have to be optional at least temporary, do we detect this case and ignore network links if it's not supplied?  Or do we just crash?  We should add code to ignore them and spit out an additional warning.